<!-- Note -->
<!-- One of our node will shutdown automatically after publishing the information, it is done to free up the memory 
The name of the node is 'node_package_info' -->

<!-- Note 2
It is by default set to update into eyantra's sheet, if want to update into our sheet, change it from the config file
The config file can be found at pkg_ros_iot_bridge/config/config_pyiot.yaml -->


<launch>

	<!-- Recording Bag File for Submission -->
	<arg name="record" default="false"/>
	<arg name="rec_name" default="task6.bag"/>
		<group if="$(arg record)">
			<node name="rosbag_record_pick" pkg="rosbag" type="record"
		   args="record -O $(find pkg_task5)/bag_files/$(arg rec_name) --chunksize=20 /eyrc/vb/ur5_1/vacuum_gripper/logical_camera/ur5_1 /eyrc/vb/ur5_2/vacuum_gripper/logical_camera/ur5_2" output="screen"/>
		</group>

	<!-- Launch Task-5 Simulation Environment in Gazebo -->
	<include file="$(find pkg_vb_sim)/launch/task6_simulation.launch" />

	<!-- Moveit Configuration for the two arms -->
	<include file="$(find pkg_vb_sim)/launch/two_ur5_move_group.launch" />

	<!-- Solution Node will come below-->

	<!-- Loading the Parameters -->
	<rosparam file="$(find pkg_ros_iot_bridge)/config/config_pyiot.yaml"/>

	<!-- Start the action server -->	
	<node name= "node_action_server_ros_iot_bridge" pkg= "pkg_ros_iot_bridge" type= "node_action_server_ros_iot_bridge.py" output= "screen"/>

	<!-- Node for Image Processing-->
	<node name= "node_package_info" pkg= "pkg_task5" type="node_package_info.py" output="screen"/>

	<!-- To Control the first arm -->
	<node name= "node_ur5_1" pkg= "pkg_task5" type="node_ur5_1.py" output="screen"/>

	<!-- To Control the second arm -->
	<node name= "node_ur5_2" pkg= "pkg_task5" type="node_ur5_2.py" output="screen"/>

</launch>
